### ISSUE TITLE
Modules set to display with login redirect page do not display automatically

### ISSUE THREAD

**Issue by greenlanegreb** [joomla/joomla-cms#23161]
### Steps to reproduce the issue

1. Create an ACL Group called "Contact Centre" under "Registered". 

2. Create a user called "Contact Centre" and set the user to be a member of "Contact Centre". 

3. Create a "Landing Page - you are logged in" article and set viewing access to "Registered" and assign it to the login redirection page in the login module. 

4. Create a "Landing Page- "Contact Centre" article and set viewing access to "Contact Centre". 

5. Create a menu and module called "Departments", link them within the module (as usual) and set it to viewing access "Registered". Set up a logged in item "Welcome" as a menu item. This should display the menu and does upon login as "Contact Centre". 

6. Create a menu and module called "Contact Centre", link within the module (as usual) and set it to viewing access "Contact Centre". Add the "Landing Page- "Contact Centre" article to this menu setting the article viewing access to "Contact Centre". Position this menu to the left (I doubt that bit matters but it has the benefit of making it more visibly obvious upon re-creation of the issue). 

7. Set the menu module "Contact Centre" to appear on the same page as the login redirect is set to (with the login redirect being a menu item on the division menu). 

### Expected result

Because the Contact Centre module item is set to ACL "Contact Centre" (that is a member of the "Registered ACL Level), the Contact Centre menu module should appear upon login since the menu module is set to display alongside the "you are logged in" menu item that is redirected to within the log in module.  

### Actual result

The actual result is the Contact Centre menu does not display unless a second item on the departments menu assigned to the "Contact Centre" ACL group is selected after login. 

In short, menus assigned to an item that is redirected to through the login module are not triggered to display. 

### System information (as much as possible)

Joomla 3.9.0. 

### Additional comments

The reason I am doing it this way is that I have multiple different divisions and I need unique menus with links within them to display dependent on different ACL access. 

I have tried to articulate this as best as I can, when you are tired after trying to fix this for 12 hours, please forgive me a lengthy step by step guide to a permissions issue is the last thing that you need.

**Comment by DavidBoggitt** [comment]
I know this doesn't address a potential bug as you describe above, but have you tried using RegularLabs' "Advanced Module Manager" ?

**Comment by greenlanegreb** [comment]
Thanks David, Yes, definitely interested. I've had a cursory glance. I notice there is a paid version and a free version. What you are suggesting, would that require the paid version or the free version? I ask because I don't trade and costs are already coming out of pocket. 

I notice there is a potential to force modules to display on strict assignment rules - how would you envisage me accomplishing the result above as I have never used Advanced Module Manager before? 

If you would be kind enough to give me some more in depth instructions, I would be happy to test it out on a copy of the existing install? The other question is, how would I go about changing my existing setup to take advantage of this - for example, could my current viewing access arrangements and viewing levels conflict with any configuration of Advanced Module Manager? 

Thanks again.

**Comment by Quy** [comment]
@greenlanegreb You may want to post your question in [Regular Labs Support Forum](https://www.regularlabs.com/forum) to get an answer.

**Comment by greenlanegreb** [comment]
Ah it turned out not to be a bug. It was just the way that Joomla does things. I messed around and found out how to sort it out. I have set up two groups that have their own menu items visible to them - IT and Business. Where I have used these, obviously people can use what they like. 

1. Install Joomla

2. Install Gantry Framework (optional) 

3. Make Sure Both Are Up to Date

4. Create User Group IT with group parent set to Registered. 

5. Create User Group Business with group parent set to Registered. 

6. Create Access Level IT - Set User Groups Having Access to "IT"

7. Create Access Level Business - Set User Groups Having Access to "Business". 

8. Create User IT - assign "registered" and "IT" in user groups tab. 

9. Create User Business assign "Registered" and "Business" in user groups tab. 

8. In Article Manager, create "Welcome - you are not signed in" page - assign viewing access to guest (no need for people logged in to see this unless they are logged out).

9. In Article Manager, create "Welcome - you are now signed in" page - assign viewing access to registered (so that all of those logged in can see it). 

10. On Main Menu,  Link main menu item to "Welcome you are not signed in". Make the menu item visible to the "guest" group (available by default).
 
11. On Main Menu, create new menu item for "Welcome - you are now signed in" - assign viewing access to registered. 

12. In default Joomla login module, set login redirect to "Welcome you are now signed in" and also ensure that the module is assigned to all pages (because users also need a logout functionality that comes with it). 

13. Create a new menu "IT" - create two test web links so that Joomla has the ability to display the menu module. Assign a module to it and set viewing access to User Role IT. The test links should also be set up as visible to the IT group. 

14. Create a new menu module "Business" - create two test web links that are distinguishable so that Joomla has the ability to display this menu module to. Set viewing access to User Role Business. The test links should also be set to only be visible to the Business group. 

15. In Menu assignment, Assign module created for "IT" to display on "You are now signed in" page linked to the main menu. 

16. In Menu assignment, Assign module created for "Business" to display on "You are now signed in" page linked to the main menu.

It would be nice to have an ability not to have to have the logged in redirect item as a main menu item and simply be able to assign modules to appear in the event of a login.. This would simplify matters significantly. Guest is also not particularly clear since there is also the public option.
